test/e2e/operator/handler_install_test.go

```diff
@@ -0,0 +1,91 @@
+package operator
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	nmstatev1beta1 "github.com/nmstate/kubernetes-nmstate/pkg/apis/nmstate/v1beta1"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	v1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	nmstate       nmstatev1beta1.NMState
+	nmstateKey    = types.NamespacedName{Namespace: "nmstate", Name: "nmstate"}
+	dsKey         = types.NamespacedName{Namespace: "nmstate", Name: "nmstate-handler"}
+	deploymentKey = types.NamespacedName{Namespace: "nmstate", Name: "nmstate-operator"}
+)
+
+func installHandler() {
+	err := framework.Global.Client.Get(context.TODO(), nmstateKey, &nmstate)
+	Expect(err).To(HaveOccurred())
+	err = framework.Global.Client.Create(context.TODO(), &nmstate, &framework.CleanupOptions{})
+	Expect(err).To(Not(HaveOccurred()))
+}
+
+func uninstallHandler() {
+	err := framework.Global.Client.Get(context.TODO(), nmstateKey, &nmstate)
+	Expect(err).To(Not(HaveOccurred()))
+	err = framework.Global.Client.Delete(context.TODO(), &nmstate, &client.DeleteOptions{})
+	Expect(err).To(Not(HaveOccurred()))
+}
+
+var _ = Describe("handler install", func() {
+	BeforeEach(func() {
+		nmstate = nmstatev1beta1.NMState{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "nmstate",
+				UID:  "12345",
+			},
+		}
+	})
+	Context("when handler not installed", func() {
```

Comment:
Context("when installed for the first time"), func() {
   BeforeEach(func(){
      installNMState()
})